dask_sql/physical/rel/custom/create_model.py

```diff
@@ -130,6 +135,13 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
         wrap_fit = kwargs.pop("wrap_fit", None)
         fit_kwargs = kwargs.pop("fit_kwargs", {})
 
+        training_df = context.sql(select)
+
+        if type(training_df) == dd.core.DataFrame:
+            model_class = cpu_classes.get(model_class, model_class)
+        elif "cudf" in str(training_df._partition_type):
+            model_class = gpu_classes.get(model_class, model_class)
```

Comment:
Nitpick but we should be able to avoid the Dask import here by inverting the conditional:

```suggestion
        if is_cudf_type(training_df):
            model_class = gpu_classes.get(model_class, model_class)
        else:
            model_class = cpu_classes.get(model_class, model_class)
```